### PR TITLE
Add Documentation

### PR DESCRIPTION
**Pull request by kavishgambhir**
Ref: #20

### CODE REVIEW

File: docs/usage/filter.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+---
+
+
+## Grayscale
+
+The `mono` parameter can be used to filter the image to Grayscale by giving it the value `000000`.
+
+| `?w=500&h=250` | `?w=500&h=250&mono=000000`|
+|:---:|:---:|
+| ![image](https://kdarkroom.herokuapp.com/sample-image.jpg?w=500&h=250) | ![image](https://kdarkroom.herokuapp.com/sample-image.jpg?w=500&h=250&mono=000000) |
```
Comment (ajatprabha) [inline docs/usage/filter.md]: Can we get this host value `kdarkroom.herokuapp.com` from some config variable?

File: website/README.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+$ yarn
+```
+2. Run your dev server:
+
+```sh
+# Start the site
+$ yarn start
+```
+
+## Directory Structure
```
Comment (ajatprabha) [inline website/README.md]: The structure doesn't match with actual files in the PR.

File: website/blog/2017-10-24-new-version-1.0.0.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+title: New Version 0.0.4
+author: Kavish Gambhir
+---
+
+This blog post will test file name parsing issues when periods are present.
```
Comment (ajatprabha) [inline website/blog/2017-10-24-new-version-1.0.0.md]: Update this to something meaningful.

File: website/siteConfig.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  },
+];
+
+const siteConfig = {
+  title: 'Darkroom', // Title for your website.
+  tagline: 'Yet Another Image Proxy',
+  githubUrl: 'https://github.com/gojek/darkroom', // Your website URL
+  baseUrl: '/darkroom/', // Base URL for your project */
+  // For github.io type URLs, you would set the url and baseUrl like:
+  url: 'https://kavishgambhir.github.io',
```
Comment (ajatprabha) [inline website/siteConfig.js]: Should be `https://gojek.io`.

File: website/README.md (hunk 2 above)
Comment (kavishgambhir) [reply website/README.md]: That is the general structure mentioned, only the names are changed for the docs

**Comment by ajatprabha** [comment]
Also, I see that you've used Heroku to deploy the Darkroom image for docs. @baskarap can help to add necessary tokens to CI secrets so that generated image can be published to heroku in the deploy stage.

File: website/README.md (hunk 2 above)
Comment (ajatprabha) [reply website/README.md]: Okay, makes sense.

File: docs/usage/filter.md (hunk 1 above)
Comment (kavishgambhir) [reply docs/usage/filter.md]: Done

File: docs/customization.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
+---
+id: customization
+title: Customization
+---
+## Customisation
```
Comment (ajatprabha) [inline docs/customization.md]: Should be in sync with the page title i.e. `Customization`

File: docs/customization.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+You may customise darkroom, for example, you may want to write a storage backend that talks to another service and gets the images.
```
Comment (ajatprabha) [inline docs/customization.md]: Change this and all other occurrences from `customise` to `customize` just to be consistent.

File: docs/usage/size.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+The size parameters allow you to resize, crop and fit-to-crop your image.
+
+## Fit
+
+Fit mode can be used to enforce crop on an image. If this is not set, the default behaviour is to resize the image while maintaing original aspect ratio. The `w` and `h` parameters should also be set, so that the crop is defined within specific image dimensions.
+
+| `?w=500&h=250&fit=crop` | `?w=500&h=250` |
+|:---:|:---:|
+| ![image](https://kdarkroom.herokuapp.com/sample-image.jpg?w=500&h=250&fit=crop) | ![image](https://kdarkroom.herokuapp.com/sample-image.jpg?w=500&h=250) |
```
Comment (ajatprabha) [inline docs/usage/size.md]: The host is still hardcoded.

File: website/blog/2016-03-11-blog-post.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
+---
```
Comment (ajatprabha) [inline website/blog/2016-03-11-blog-post.md]: Please update this dummy post to have some meaningful data.

File: website/blog/2017-04-10-blog-post-two.md [hunk 9]
```diff
@@ -0,0 +1,18 @@
+---
```
Comment (ajatprabha) [inline website/blog/2017-04-10-blog-post-two.md]: You can remove this file.

File: website/blog/2017-09-25-testing-rss.md [hunk 10]
```diff
@@ -0,0 +1,11 @@
+---
```
Comment (ajatprabha) [inline website/blog/2017-09-25-testing-rss.md]: Remove this fie.

File: website/blog/2017-09-26-adding-rss.md [hunk 11]
```diff
@@ -0,0 +1,10 @@
+---
```
Comment (ajatprabha) [inline website/blog/2017-09-26-adding-rss.md]: Remove this file.

File: website/siteConfig.js [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+  tagline: 'Yet Another Image Proxy',
+  githubUrl: 'https://github.com/gojek/darkroom', // Your website URL
+  baseUrl: '/darkroom/', // Base URL for your project */
+  // For github.io type URLs, you would set the url and baseUrl like:
+  url: 'https://gojek.io',
+  //   baseUrl: '/test-site/',
+
+  // Used for publishing and more
+  projectName: 'darkroom',
+  organizationName: 'kavishgambhir',
```
Comment (ajatprabha) [inline website/siteConfig.js]: This should be `gojek`

**Comment by ajatprabha** [comment]
Overall this looks good, left a few comments, please address them. Also, this PR requires two environment variables viz. `GITHUB_USER` and `GITHUB_TOKEN`, latter is already there in the secrets. CMIIW @kavishgambhir 

I think `GITHUB_USER` should be `irfn`, since the token corresponds to his account. @irfn Please add the environment variable `GITHUB_USER=irfn` as a non-secret variable and let us know so that we can merge this when it is ready.

**Comment by kavishgambhir** [comment]
@ajatprabha Moved `Dockerfile => build/Dockerfile.docs` and retained `.dockerignore` and `docker-compose.yml` in the root as I think they should be in the root of the project.

**Comment by ajatprabha** [comment]
Okay, cool. Can you update `.travis.yml` to build the docs only on master branch? Also, add the `GITHUB_USER` in the CI YAML for now. We can remove it later when it is added to the CI env variables.

File: website/sidebars.json [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+          "usage/rotate",
+          "usage/filter"
+        ]
+      },
+      "customization",
+      "contribute"
+    ]
+  },
+  "docs-other": {
+    "First Category": ["doc5"]
```
Comment (ajatprabha) [inline website/sidebars.json]: Remove this!

**Comment by ajatprabha** [comment]
This looks good, merging now to test if it actually works on `master`.

**Comment by ajatprabha** [comment]
It looks as expected. Can we review this and see if everything is fine?
One thing that remains is deploying the docker image to Heroku from TravisCI so that docs are in sync with deployed version.
cc: @kavishgambhir @baskarap @adalberht

**Pull request merged**